src/main/java/lotto/domain/Lotto.java

```diff
@@ -19,13 +16,21 @@ public Lotto(List<LottoNumber> lottoNumbers) {
         this.numbers = Collections.unmodifiableList(lottoNumbers);
     }
 
-    public int winningNumberMatchCount(Lotto lotto) {
+    public WinningNumberMatchResult winningNumberMatch(Lotto winningNumbers, LottoNumber bonusNumber) {
         int winningNumberMatchCount = 0;
 
-        for (LottoNumber winningNumber : lotto.numbers) {
+        for (LottoNumber winningNumber : winningNumbers.numbers) {
             winningNumberMatchCount += winningNumberMatchCount(winningNumber.getNumber());
         }
-        return winningNumberMatchCount;
+        return new WinningNumberMatchResult(winningNumberMatchCount, isBonusNumberMatch(bonusNumber));
+    }
+
+    public List<LottoNumber> lottoNumbers() {
+        return new ArrayList<>(this.numbers);
+    }
+
+    private boolean isBonusNumberMatch(LottoNumber bonusNumber) {
+        return this.numbers.contains(bonusNumber);
```

Comment:
메서드 명과 파라미터 명을 아래와 같이 변경해보는건 어떨까요?
```suggestion
    private boolean contains(LottoNumber lottoNumber) {
        return this.numbers.contains(lottoNumber);
```

Reply:
제가 작성한 메소드 시그니처는 보너스 숫자에 대한 책임으로 한정 되지만 변경해주신 걸 보면 로또에 일치하는 번호가 있는지 확인하는 것이니 훨씬 범용적으로 사용할 수 있겠네요!

Reply:
제가 작성한 메소드 시그니처는 보너스 숫자에 대한 책임으로 한정 되지만 변경해주신 걸 보면 로또에 일치하는 번호가 있는지 확인하는 것이니 훨씬 범용적으로 사용할 수 있겠네요!